keylime/tenant_webapp.py

```diff
@@ -528,105 +538,106 @@ def post(self):
             logger.warning('POST returning 500 response. Tenant error: %s'%str(e))
             common.echo_json_response(self, 500, "Request failure", str(e))
             return
-        
+
         common.echo_json_response(self, 200, "Success")
-    
+
     def put(self):
-        """This method handles the PUT requests to add agents to the Cloud Verifier. 
-         
+        """This method handles the PUT requests to add agents to the Cloud Verifier.
+
         Currently, only agents resources are available for PUTing, i.e. /agents. All other PUT uri's will return errors.
         """
-        
+
         rest_params = common.get_restful_params(self.request.uri)
         if rest_params is None:
             common.echo_json_response(self, 405, "Not Implemented: Use /agents/ interface")
             return
-        
+
         if "agents" not in rest_params:
             common.echo_json_response(self, 400, "uri not supported")
             logger.warning('PUT returning 400 response. uri not supported: ' + self.request.path)
             return
-        
+
         agent_id = rest_params["agents"]
-        
-        # let Tenant do dirty work of reactivating agent 
+
+        # let Tenant do dirty work of reactivating agent
         mytenant = tenant.Tenant()
         mytenant.agent_uuid = agent_id
         mytenant.do_cvreactivate()
-        
+
         common.echo_json_response(self, 200, "Success")
 
 
 def parse_data_uri(data_uri):
     if data_uri is None:
         return None
-    
+
     data = []
-    
+
     dataset_uris = data_uri.split("\n")
     for uri in dataset_uris:
         fpos = uri.find(",")
-        if fpos == -1: 
+        if fpos == -1:
             return None
-        
+
         try:
-            data.append(base64.b64decode(uri[fpos:]))
+            data.append(base64.b64decode(uri[fpos:])).decode('utf-8'))
```

Comment:
This got messed up somehow -- it should be: 

````python
data.append(base64.b64decode(uri[fpos:]).decode('utf-8'))
````

It looks like double parenthesis got added in here: `uri[fpos:])).decode(`

Reply:
done!